src/include/splash/AttributeInfo.hpp

```diff
@@ -0,0 +1,133 @@
+        size_t getMemSize() throw(DCException);
+        /** Return the CollectionType. Might be ColTypeUnknown */
+        const CollectionType& getType() throw(DCException);
+        /** Return the size in each dimension */
+        Dimensions getDims() throw(DCException);
+        /** Return the number of dimensions */
+        uint32_t getNDims() throw(DCException);
+        /** Return whether this is a scalar value */
+        bool isScalar()  throw(DCException){ return getNDims() == 1 && getDims().getScalarSize() == 1; }
+        /** Return true, if the attribute has a variable size in which case reading
+         *  it will only return its pointer */
+        bool isVarSize() throw(DCException);
+        /** Read the name of the attribute. Returns an empty string on error */
+        std::string readName();
+
+        /**
+         * Read the data of this attribute into the buffer of the given type.
+         * Data conversion may occur, if required.
+         * See the HDF5 manual for details (6.10. Data Transfer: Datatype Conversion and Selection)
+         * Throws an exception if the attribute could not be read or converted
+         *
+         * @param colType Type of the buffer
+         * @param buf     Pointer to buffer. Size must be equal to \ref getMemSize
+         */
+        void read(const CollectionType& colType, void* buf) throw(DCException);
+
+        /**
+         * Read the data of this attribute into the buffer of the given type.
+         * Data conversion may occur, if required.
+         * See the HDF5 manual for details (6.10. Data Transfer: Datatype Conversion and Selection)
+         * Does not throw exception, so it can be used for trying different types
+         *
+         * @param colType Type of the buffer
+         * @param buf     Pointer to buffer. Size must be equal to \ref getMemSize
+         * @return True on success, else false
+         */
+        bool readNoThrow(const CollectionType& colType, void* buf);
+
+        /**
+         * Read the data of this attribute into the buffer of the given size.
+         * No data conversion occurs, so the type as found in the file is used.
+         * Throws an exception if the attribute could not be read or the buffer is to small.
+         * Warning: Might be unsafe if you are not the author of the HDF5 file or the data type
+         *          differs from the expected one. Check the meta-data and/or pass the
+         *          CollectionType to make sure you get what you expect.
+         *
+         * @param buf     Pointer to buffer
+         * @param bufSize Size of the buffer
+         */
+        void read(void* buf, size_t bufSize) throw(DCException);
+    private:
+
+        std::string getExceptionString(const std::string& msg);
+        void loadType() throw(DCException);
+        void loadSpace() throw(DCException);
+        
+        /** Reference counter for the attribute */
+        unsigned* refCt_;
```

Comment:
However I felt that this looks wrong to have it in this class. Moved it to the IdWrapper via policies loosely based on the SmartPtr from the Loki library.